File: src/token.py
# token.py - Reed Foster
# A class for Tokens

class Token(object):
    def __init__(self, type, value):
        self.type = type
        self.value = value

    def __str__(self):
        return 'Token({type}, "{value}")'.format(type=self.type, value=self.value.__str__())

File: java/src/com/foster/cdl/Token.java
/*
Token.java - Reed Foster
Class for Token objects
*/

package com.foster.cdl;

public class Token
{   
    public final Tokentype type;
    public final String value;
    
    Token(Tokentype type, String value)
    {
        this.type = type;
        this.value = value;
    }
    
    public String toString()
    {
        return String.format("(%s, \"%s\")", this.type, this.value);
    }
}
